### ISSUE TITLE
Creating "Video Plugin" #2

### ISSUE THREAD

**Issue by starkiJi** [Scirra/Construct-3-bugs#5742]
**Problem description**
Hello, I'm using google translator. Old problem... - https://github.com/Scirra/Construct-3-bugs/issues/5524 

**Can you make 1) new video plugin №2? or 2) buy another codec? Or 3) buy a player and build it into the engine?**

My concept of the game involves a large number of video scenes, but they slow down in 1920-1080 resolution on medium devices. I have to decrease the size of their resolution, but then the videos will be blurry....

The funny thing is that I downloaded a regular player from the Androyd store and it plays video in 2560-1440 resolution smoothly, and I only need 1920-1080, but construct 3 cannot cope with this task yet.

Even if the video weighs less than 1 megabyte, there are jerks - 

**Attach a .c3p**
https://drive.google.com/file/d/19nP8tKn_aWY7h87d4BPJMnOFyP6DfJg5/view?usp=sharing

Sorry to bother you with such a minor thing, but perhaps there is a theoretical hypothetical solution to the problem. I am a longtime buyer of C2 and C3 and I may need the functionality of your engine to create other types of games with video inserts, but so far they only work on a computer

**Comment by AshleyScirra** [comment]
I'm not clear what you're doing here exactly - does this affect desktop devices? Android devices in the browser? Or Android devices in an APK?

Regardless, the video playback is handled by the browser, and I am afraid I doubt we can improve it any further than the work already done in issue #5524.

**Comment by starkiJi** [comment]
Sorry for the incorrect description, I meant that even on devices that are about 7 years old, braking occurs., which surprises me, on a tablet. I understand that you cannot improve this right now, but I wanted to know if there are any other theoretical options? sew a new video player into the engine, if possible ....

APK file, where the video weighs 700 kilobytes - https://drive.google.com/file/d/11cjfftzgi1O2-7WKwJC5D9QJiRnIJ6QF/view?usp=sharing

**Comment by AshleyScirra** [comment]
I'm assuming you are talking about Android apps (e.g. APKs)? You still haven't described exactly what you are doing.

I'm afraid I don't believe it is feasible to make any changes. The browser is ultimately responsible for video playback, and we have already made all the improvements we can in issue #5524. Future improvements may be possible, but they are currently blocked on Google making further improvements to Chrome.

Also note the file size has nothing to do with it. The performance cost is based on the size of the video. For example 1920x1080 pixels with 8-bit RGB channels 6mb for a single frame. at 30 FPS that means a bandwidth of 186mb per second. At 60 FPS it would be 373mb per second. If the compositor has to copy that twice to display a single frame, it's around 746mb per second. This is a high performance cost and old devices can struggle with that. Newer devices should be OK though.

**Comment by starkiJi** [comment]
@AshleyScirra I'm sorry, that's the last question, I hope.
I launched C3 on my old tablet and opened the downloaded video in the video preview mode (not to be confused with the general preview of the entire scene, just the video after import) and there my video has a size of 2560 -1440 (and I wanted all 1920-1080) and it does not seem to slow down, everything is cool. The question is, in what then is this video played, not in the browser? This may not help in the future to solve my problem .... can this method be used to make "video plug-in #2"? 

video link - https://drive.google.com/file/d/1x3CXe9DXMCgwnE59A9LXwRO-QZoj4MUd/view?usp=sharing

**Comment by starkiJi** [comment]
Preview of the whole scene - in it, the video in size 1280-1024 slows down, and the preview of a specific video in a folder after importing the video, even if the video has a video size of 2560-1440, works perfectly. This is what I need. Testing was on the old tablet of 2016

**Comment by starkiJi** [comment]
illustration - https://ibb.co/K654cTb

**Comment by starkiJi** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=1329803 - And if it's not difficult to answer in this report what needs to be fixed in order for the video 1920-1080 to work smoothly, as if playing through a video player, thanks.

**Comment by AshleyScirra** [comment]
It's not feasible to make a whole new video plugin.

We already have some issues filed with Google, and if they fix them we may be able to improve performance. But we have to wait for Google first.
